app/Platform/Controllers/PlayerCompletionProgressController.php

```diff
@@ -0,0 +1,319 @@
+            $userGamesList,
+            $userSiteAwards
+        );
+
+        // If the user is only wanting to see results for a specific console,
+        // filter out every game and award that doesn't have that console ID.
+        if ($targetSystemId) {
+            [$userGamesList, $userSiteAwards] = $this->playerProgressionService->useSystemId(
+                $targetSystemId,
+                $userGamesList,
+                $userSiteAwards
+            );
+        }
+
+        // Remove invalid consoles and add some awards metadata to each entity.
+        $filteredAndJoinedGamesList = $this->playerProgressionService->filterAndJoinGames(
+            $userGamesList,
+            $userSiteAwards,
+        );
+        $primaryCountsMetrics = $this->playerProgressionService->buildPrimaryCountsMetrics($filteredAndJoinedGamesList);
+
+        // The user may only be wanting to see a certain subset of games, such
+        // as only mastered games, or only unawarded games. Perform this filtering now.
+        if ($targetGameStatus) {
+            $filteredAndJoinedGamesList = $this->useGameStatusFilter(
+                $targetGameStatus,
+                $filteredAndJoinedGamesList
+            );
+        }
+
+        $milestones = $this->buildMilestones($filteredAndJoinedGamesList);
+
+        $filteredAndJoinedGamesList = $this->useSortedList($filteredAndJoinedGamesList, $sortOrder);
+
+        $totalInList = count($filteredAndJoinedGamesList);
+        $totalPages = ceil($totalInList / $this->pageSize);
+
+        // Ensure the user isn't trying to go out of bounds.
+        if ($currentPage < 1 || $currentPage > $totalPages) {
+            $currentPage = 1;
+        }
+
+        $paginatedGamesList = array_slice(
+            $filteredAndJoinedGamesList,
+            ($currentPage - 1) * $this->pageSize,
+            $this->pageSize,
+        );
+
+        return view('platform.completion-progress-page', [
+            'allAvailableConsoleIds' => $allAvailableConsoleIds,
+            'completedGamesList' => $paginatedGamesList,
+            'currentPage' => $currentPage,
+            'isFiltering' => $targetGameStatus || $targetSystemId,
+            'me' => $me,
+            'milestones' => $milestones,
+            'primaryCountsMetrics' => $primaryCountsMetrics,
+            'selectedConsoleId' => $targetSystemId,
+            'selectedSortOrder' => $sortOrder,
+            'selectedStatus' => $targetGameStatus,
+            'siteAwards' => $userSiteAwards,
+            'totalInList' => $totalInList,
+            'totalPages' => $totalPages,
+            'user' => $foundTargetUser,
+        ]);
+    }
+
+    private function useSortedList(array $filteredAndJoinedGamesList, string $sortOrder): array
+    {
+        if ($sortOrder === 'unlock_date') {
+            usort($filteredAndJoinedGamesList, function ($a, $b) {
+                return strtotime($b['MostRecentWonDate']) - strtotime($a['MostRecentWonDate']);
```

Comment:
I apparently have awards without `MostRecentWonDate`s:
```
strtotime(): Argument #1 ($datetime) must be of type string, null given
```
I can no longer open the page for most of the consoles in my list.

Reply:
OK. Figured it out.

With `FEATURE_AGGREGATE_QUERIES` enabled, I get back a null `MostRecentWonDate` from `pg.last_unlock_at AS MostRecentWonDate`.

With `FEATURE_AGGREGATE_QUERIES` disabled, I get back an empty string for `MostRecentWonDate` from `MAX(aw.Date) AS MostRecentWonDate`.

`strtotime` appears to work with empty string, but not with null.

Reply:
Really good find, thank you.

In https://github.com/RetroAchievements/RAWeb/pull/1833/commits/907562b7b77e1ebdf28e41d42d548489aeb2bf98 I've added some nullish coalescing fallbacks:

```php
if ($sortOrder === 'unlock_date') {
    usort($filteredAndJoinedGamesList, function ($a, $b) {
        return strtotime($b['MostRecentWonDate'] ?? '') - strtotime($a['MostRecentWonDate'] ?? '');
    });
}

if ($sortOrder === '-unlock_date') {
    usort($filteredAndJoinedGamesList, function ($a, $b) {
        return strtotime($a['MostRecentWonDate'] ?? '') - strtotime($b['MostRecentWonDate'] ?? '');
    });
}
```